### ISSUE TITLE
Could not connect to server.

### ISSUE THREAD

**Issue by alvinvarghese** [Lejdborg/NMSSH#157]
let session : NMSSHSession = NMSSHSession(host: "", port: 22, andUsername: "")
        
        if session.connectToAgent()
        {
            if session.connected
            {
            session.authenticateByPassword("")
            
            if session.authorized {
                
                print("Authentication succeeded")
            }
            }
        }

I was using this code, to connect to server. It was not working, so in order to check whats happening i put a breakpoint in "if session.connectToAgent()" then it redirected me to this "supportedAuthenticationMethods" page, and never executed. Here this "userauthlist" returned NULL thats why it was not working i guess. 

- (NSArray *)supportedAuthenticationMethods {
    char *userauthlist = libssh2_userauth_list(self.session, [self.username UTF8String],
                                               (unsigned int)strlen([self.username UTF8String]));
    if (userauthlist == NULL){
        NMSSHLogInfo(@"Failed to get authentication method for host %@:%@", self.host, self.port);
        return nil;
    }

    NSString *authList = [NSString stringWithCString:userauthlist encoding:NSUTF8StringEncoding];
    NMSSHLogVerbose(@"User auth list: %@", authList);

    return [authList componentsSeparatedByString:@","];
}

- (BOOL)supportsAuthenticationMethod:(NSString *)method {
    return [[self supportedAuthenticationMethods] containsObject:method];
}

Please help me out why is this happening ?. Xcode 7, Swift 2.0, iOS 9.0. Integrated NMSSH via CocoaPods.

**Comment by Lejdborg** [comment]
In your example you use no host or username. Is it redacted or have you not configured any?

Is the server open for SSH connections on port 22?

**Comment by alvinvarghese** [comment]
No, Actually i used host name, user name and password. Just to tell what happened i removed those values from here.

hostName : sftp.dewdrive.com, port : 22. I am trying to connect to SFTP

**Comment by Frugghi** [comment]
[`libssh2_userauth_list`](http://www.libssh2.org/libssh2_userauth_list.html) returns NULL on failure or when the server supports *SSH_USERAUTH_NONE* (no authentication). Maybe this is your case?

**Comment by alvinvarghese** [comment]
I am trying to connect to dewdrive.com, which is hosted in uk2.net as per the whois details. I don't know what you exactly meant by "Maybe this is your case?". There is no problem related with server, username and password. Because i tried connecting to server just a moment ago using a MacApp. So could you please give what else i miss in my iOS project or in my server side ??

**Comment by Frugghi** [comment]
Does your server support authetication method "NONE"?

**Comment by alvinvarghese** [comment]
I don't know about this, i should ask this to uk2.net technical team i think.
